### PR TITLE
[Bugfix] Filter by polygon: table could be a query

### PR DESCRIPTION
**Pull request by rldhont**
The layer filtered by polygon could be a query `_features_ids_with_sql_query`

The polygon layer  could be a query `_polygon_for_groups_with_sql_query`

### CODE REVIEW

**Comment by Gustry** [comment]
I was curious how the tests could pass.

If the code is not really tested before the PR, please specify it, otherwise, we were going to make some 500 errors if merged too quickly.

image

`isEmpty` does not exist in Python, and it's `startswith`.

I want to check how CI could be OK, and also other checks on this PR later

**Comment by rldhont** [comment]
> 
> `isEmpty`, `startsWith`, `endsWith` does not exist in Python
> 

`isEmpty`, `startsWith`, `endsWith` are `QString` methods not python string ones. I was thinking that is not converted.

I think there is no tests to check build sql.

**Comment by rldhont** [comment]
I have updated the code and added a test.

File: lizmap_server/filter_by_polygon.py [hunk 1]
```diff
@@ -495,3 +494,14 @@ def _format_qgis_expression_relationship(
     {current_geometry}
 )"""
         return expression
+
+    @classmethod
+    def _format_table_name(cls, uri: QgsDataSourceUri) -> str:
+        # is the datasource a query or a table ?
```
Comment (Gustry) [inline lizmap_server/filter_by_polygon.py:500]:
```suggestion
        """ Format the table name according to the URI. """
```

**Review by Gustry — APPROVED** [review]
LGTM
Perfect, no code duplication, single `static` or `class` method, which is unit-testable ;-)

> isEmpty, startsWith, endsWith are QString methods not python string ones

While `QString` belongs to the [QtCore package](https://doc.qt.io/qt-5/qstring.html), it does not exist in Python. It's absolutely not possible to have a QString object in PyQt.

```
from qgis.PyQt.QtCore import QString
Traceback (most recent call last):
  File "/usr/lib/python3.12/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
ImportError: cannot import name 'QString' from 'qgis.PyQt.QtCore' (/usr/lib/python3/dist-packages/qgis/PyQt/QtCore.py)
```

**Comment by rldhont** [comment]
> 
> While `QString` belongs to the [QtCore package](https://doc.qt.io/qt-5/qstring.html), it does not exist in Python. It's absolutely not possible to have a QString object in PyQt.
> 
I found the same information after your review. Thanks to point it.

**Review by Gustry — APPROVED** [review]

**Pull request merged**